hack/tools.go

```diff
@@ -0,0 +1,25 @@
+// +build tools
+
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tools
+
+// This package imports things required by this repository, to force `go mod` to see them as dependencies
+import (
+	_ "k8s.io/code-generator"
```

Comment:
I thought you needed to list individual code-generators, do you not?

Should this list the main package for our code-generator as well?

Reply:
This is how k8s/sample-controller does - https://github.com/kubernetes/sample-controller/blob/master/hack/tools.go#L22

Reply:
I don't think this is working, I do think you have to include the individual code-generators

Reply:
it turns out you need both.

Reply:
Hmm, they are already imported in https://github.com/kubernetes/code-generator/blob/master/tools.go#L24-L34. I think it should be enough for Go modules?